pkg/trace/obfuscate/sql.go

```diff
@@ -105,11 +105,39 @@ func (f *replaceFilter) Filter(token, lastToken TokenKind, buffer []byte) (token
 	switch token {
 	case String, Number, Null, Variable, PreparedStatement, BooleanLiteral, EscapeSequence:
 		return FilteredGroupable, []byte("?"), nil
+	case '?':
+		// Cases like 'ARRAY [ ?, ? ]' should be collapsed into 'ARRAY [ ? ]'
+		return FilteredGroupable, []byte("?"), nil
+	case Table:
+		if config.HasFeature("normalize_sql_tables") {
+			return token, f.replaceTableDigits(buffer), nil
+		}
+		fallthrough
 	default:
 		return token, buffer, nil
 	}
 }
 
+// replaceTableDigits replaces consecutive sequences of digits with '?',
+// example: "jobs_2020_1597876964" --> "jobs_?_?"
+func (f *replaceFilter) replaceTableDigits(buffer []byte) []byte {
```

Comment:
I think we can reduce it to `replaceDigits` or `obfuscateDigits` since it only replaces digits in a string, not in a table. What do you think?

Reply:
Sure, the name was to imply that it was only to be used on tables, but if someone wants to reuse it on something else then I don't see a problem with that.

Reply:
Got it. The principle for naming things should be "what the thing does" not "where it is used". It makes it easier to piece things together (imho). Thanks for accepting the suggestion.